### PR TITLE
Make the suggestion menu type configurable

### PR DESCRIPTION
**Pull request by elazarcoh**
I found little time to put it together. 

I started to look at how the other configurations are used, so I can integrate the new config better, but it look like other configurations are related more to the server and not the client. I ended up writing it in the most local way. You know the extension much better, is there another way you want me to integrate it? If so, just point me to an example, if there is any. I can figure it out myself from there, probably.

Thanks!

### CODE REVIEW

File: packages/client/src/commands.ts [hunk 1]
```diff
@@ -563,6 +563,12 @@ async function actionJumpToSpellingError(which: 'next' | 'previous', suggest: bo
     editor.selection = new Selection(range.start, range.end);
 
     if (suggest) {
-        await commands.executeCommand('cSpell.suggestSpellingCorrections');
+        type SuggestionMenu = 'quickPick' | 'quickFix';
```
[the remark this answers is not in the document]
Comment (elazarcoh) [reply packages/client/src/commands.ts:566]: > ```ts
> suggestionMenuCommand
> ```

I think it less user friendly. It's harder to understand what each one means.

File: packages/client/src/commands.ts [hunk 2]
```diff
@@ -563,6 +563,12 @@ async function actionJumpToSpellingError(which: 'next' | 'previous', suggest: bo
...
+        const menu = workspace.getConfiguration('cSpell').get<SuggestionMenu>('suggestionMenu');
```
[the remark this answers is not in the document]
Comment (elazarcoh) [reply packages/client/src/commands.ts]: 👍

**Comment by elazarcoh** [comment]
Updated to code to use the settings as you said.

Do this method of generating the configuration from code is a standalone npm library? I might want to use it in my extensions as well. It's really nice way to handle all those redundant duplication as type castings.

**Comment by Jason3S** [comment]
> Updated to code to use the settings as you said.

Thank you. It looks good!

> Do this method of generating the configuration from code is a standalone npm library? I might want to use it in my extensions as well. It's really nice way to handle all those redundant duplication as type castings.

I wrote it myself. I was getting frustrated trying to keep everything in sync. The code is not so clean. Pulling it out into its own standalone library could be useful and would force a code clean-up.

Unrelated note:

I have also mocked VS Code in order to do unit tests. Code coverage in the client is low since it was so hard to do 
unit tests before writing the mock. I only added the mock recently. 
I might publish the Jest VS Code Mock soon, since I think others might find it useful. 
Getting good code coverage for the extension is hard.  
To check it out, from the `client` folder: `yarn jest --coverage`. 
Then open `coverage/lcov-report/index.html`. Slowly things are getting better, but it will take time.

**Comment by Jason3S** [comment]
**Some checks were not successful**
Looks like `packages/_server/spell-checker-config.schema.json` didn't get added to the diff.

```
Run git --no-pager diff --compact-summary --exit-code
 packages/_server/spell-checker-config.schema.json | 14 ++++++++++++++
 1 file changed, 14 insertions(+)
Error: Process completed with exit code 1.
```

**Comment by Jason3S** [comment]
Thank you for doing this. It is these little things that help improve the user experience.

**Comment by elazarcoh** [comment]
> I wrote it myself. I was getting frustrated trying to keep everything in sync. The code is not so clean. Pulling it out into its own standalone library could be useful and would force a code clean-up.

So if that OK with you, I would like to extract it out and make it a standalone. It might take me a while to get to it, though. 

I'll add you to the repo I'll open for it, if you wish it.

As I see it, it even might become a standard for extension developers...

**Pull request merged**